xc/device_common.ml

```diff
@@ -238,6 +238,14 @@ let readdir ~xs d = try xs.Xs.directory d with Xs_protocol.Enoent _ -> []
 let to_list ys = List.concat (List.map Opt.to_list ys)
 let list_kinds ~xs dir = to_list (List.map parse_kind (readdir ~xs dir))
 
+let filter_on_devids ~xs ?for_devids dir =
+  match for_devids with
+  | None -> List.filter_map parse_int (readdir ~xs dir)
```

Comment:
The original does `to_list (List.map`. Why change?

Reply:
It looks like `to_list` will perform List.concat  on a list of list.
But I think there should be no list of list here.
And `List.filter_map` should take same function on that.

Reply:
If you are just factoring out a function, then please don't change it unless there is a very good reason. And if you do, mention the change explicitly. Otherwise it just causes confusion and it increases the possibility of mistakes and bugs.